controllers/serviceaccount_controller_intg_test.go

```diff
@@ -59,14 +59,21 @@ var _ = Describe("ProviderServiceAccount controller integration tests", func() {
 				helpers.CreateAndWait(ctx, intCtx.Client, intCtx.KubeconfigSecret)
 			})
 
+			By("Verifying that the guest cluster client works")
+			guestClient, err := tracker.GetClient(ctx, client.ObjectKeyFromObject(intCtx.Cluster))
+			Expect(err).ToNot(HaveOccurred())
+			// Note: Create a Service informer, so the test later doesn't fail if this doesn't work.
+			Expect(guestClient.List(ctx, &corev1.ServiceList{}, client.InNamespace(metav1.NamespaceDefault))).To(Succeed())
```

Comment:
Saw some errors in the logs. Not sure if the test failures are due to envtest not coming up (probalby not, but shouldn't hurt to have this explicit check here)